mittab/apps/tab/archive.py

```diff
@@ -0,0 +1,148 @@
+from xml.etree.ElementTree import Element, SubElement, tostring
+
+from mittab.apps.tab.models import School, Team, Judge, Room, Round
+
+DEBATE_ID_PREFIX = "D"
+ROOM_ID_PREFIX = "V"
+SPEAKER_ID_PREFIX = "S"
+TEAM_ID_PREFIX = "T"
+JUDGE_ID_PREFIX = "A"
+SCHOOL_ID_PREFIX = "I"
+SPEAKER_STATUS_PREFIX = "SC"
+
+
+class ArchiveExporter:
+
+    def __init__(self, name):
+        self.name = name
+        self.root = None
+
+    def export_tournament(self):
+        self.root = Element("tournament", {"name": self.name})
+
+        self.add_rounds()
+        self.add_participants()
+        self.add_schools()
+        self.add_rooms()
+        self.add_categories()
+
+        return tostring(self.root)
+
+    def add_rounds(self):
+        pos = ["pm", "lo", "mg", "mo"]
+        qs = Round.objects.all().order_by("round_number").prefetch_related(
+            "judges", "roundstats_set")
+
+        cur_round = None
+        r_tag = None
+        for debate in qs:
+            if debate.round_number != cur_round:
+                r_tag = SubElement(self.root, "round", {
+                    "name": "Round %s" % debate.round_number
+                })
+                cur_round = debate.round_number
+
+            adjs = " ".join(
+                [JUDGE_ID_PREFIX + str(j.id) for j in debate.judges.all()])
+            d_tag = SubElement(r_tag, "debate", {
+                "id": DEBATE_ID_PREFIX + str(debate.id),
+                "chair": JUDGE_ID_PREFIX + str(debate.chair_id),
+                "adjudicators": adjs,
+                "venue": ROOM_ID_PREFIX + str(debate.room_id)
+            })
+
+            gov_tag = SubElement(d_tag, "side", {
+                "team": TEAM_ID_PREFIX + str(debate.gov_team_id)
+            })
+            opp_tag = SubElement(d_tag, "side", {
+                "team": TEAM_ID_PREFIX + str(debate.opp_team_id)
+            })
+
+            if debate.victor == Round.GOV_VIA_FORFEIT or \
+                    debate.victor == Round.ALL_DROP:
+                opp_tag.set("forfeit", "True")
+            elif debate.victor == Round.OPP_VIA_FORFEIT or \
+                    debate.victor == Round.ALL_DROP:
+                gov_tag.set("forfeit", "True")
+
+            team_points = [0, 0] # Gov, Opp
+            stats = sorted(list(debate.roundstats_set.all().values()),
+                           key=lambda x: pos.index(x["debater_role"]))
+            for i, speech in enumerate(stats):
```

Comment:
I think we may only want to do this in the case where the round result is `GOV_WIN` or `OPP_WIN` -- there's no application logic which prevents speaks from being associated with `ALL_WIN` rounds, yet those speaks are ignored (and auto-calculated) for tabulation